front/lib/plans/subscription.ts

```diff
@@ -0,0 +1,247 @@
+import { Op } from "sequelize";
+
+import { front_sequelize, ModelId } from "@app/lib/databases";
+import { Plan, Subscription } from "@app/lib/models";
+import {
+  FREE_TEST_PLAN_DATA,
+  FREE_UPGRADED_PLAN_CODE,
+  PlanAttributes,
+} from "@app/lib/plans/free_plans";
+import { generateModelSId } from "@app/lib/utils";
+import logger from "@app/logger/logger";
+import { SubscribedPlanType } from "@app/types/user";
+
+export const getActiveWorkspacePlan = async ({
+  workspaceModelId,
+}: {
+  workspaceModelId: ModelId | null;
+}): Promise<SubscribedPlanType> => {
+  let activeSubscription: Subscription | null = null;
+  const today = new Date();
+  today.setHours(0, 0, 0, 0);
+
+  if (workspaceModelId) {
+    activeSubscription = await Subscription.findOne({
+      where: {
+        workspaceId: workspaceModelId,
+        startDate: { [Op.lte]: today },
+        [Op.or]: [{ endDate: { [Op.gt]: today } }, { endDate: null }],
+      },
+    });
+  }
+
+  // Default values when no subscription
+  let plan: PlanAttributes = FREE_TEST_PLAN_DATA;
+  let startDate = today;
+  let endDate = null;
+
+  if (activeSubscription) {
+    const subscribedPlan = await Plan.findOne({
+      where: { id: activeSubscription.planId },
+    });
+    startDate = activeSubscription.startDate;
+    endDate = activeSubscription.endDate;
+    if (subscribedPlan) {
+      plan = subscribedPlan;
+    } else {
+      logger.error(
+        {
+          workspaceModelId,
+          activeSubscription,
+        },
+        "Cannot find plan for active subscription. Will use limits of FREE_TEST_PLAN instead. Please check and fix."
+      );
+    }
+  }
+
+  return {
+    code: plan.code,
+    name: plan.name,
+    startDate: startDate?.getTime(),
+    endDate: endDate?.getTime() || null,
+    limits: {
+      assistant: {
+        isSlackBotAllowed: plan.isSlackbotAllowed,
+        maxWeeklyMessages: plan.maxWeeklyMessages,
+      },
+      managedDataSources: {
+        isSlackAllowed: plan.isManagedSlackAllowed,
+        isNotionAllowed: plan.isManagedNotionAllowed,
+        isGoogleDriveAllowed: plan.isManagedGoogleDriveAllowed,
+        isGithubAllowed: plan.isManagedGithubAllowed,
+      },
+      staticDataSources: {
+        count: plan.maxNbStaticDataSources,
+        documents: {
+          count: plan.maxNbStaticDocuments,
+          sizeMb: plan.maxSizeStaticDataSources,
+        },
+      },
+      users: {
+        maxUsers: plan.maxUsersInWorkspace,
+      },
+    },
+  };
+};
+
+/**
+ * Internal function to subscribe to the default FREE_TEST_PLAN.
+ * This is the only plan without a database entry: no need to create a subscription, we just end the active one if any.
+ */
+export const internalSubscribeWorkspaceToFreeTestPlan = async ({
+  workspaceModelId,
+}: {
+  workspaceModelId: ModelId;
```

Comment:
WorkspaceType here please

Reply:
Problem is we can this either in Poké, in Cli, or on Login and for each case we have a Workspace object not a workspace type. This is why I called it `internal` 😕

Reply:
Fixed in the other PR -> I'm passing the sId and I refetch the workspace.